apps/web/src/ee/clerk/providers/EnterpriseAuthProvider.tsx

```diff
@@ -147,27 +147,55 @@ export const EnterpriseAuthProvider = ({ children }: { children: React.ReactNode
-  _id: clerkUser.id,
-  firstName: clerkUser.firstName,
-  lastName: clerkUser.lastName,
-  email: clerkUser.emailAddresses[0].emailAddress,
-  profilePicture: clerkUser.imageUrl,
-  createdAt: clerkUser.createdAt?.toString() ?? '',
-  showOnBoarding: clerkUser.publicMetadata.showOnBoarding,
-  showOnBoardingTour: clerkUser.publicMetadata.showOnBoardingTour,
-  servicesHashes: clerkUser.publicMetadata.servicesHashes,
-  jobTitle: clerkUser.publicMetadata.jobTitle,
-  hasPassword: clerkUser.passwordEnabled,
-});
-
-const toOrganizationEntity = (clerkOrganization: OrganizationResource): IOrganizationEntity => ({
-  _id: clerkOrganization.id,
-  name: clerkOrganization.name,
-  createdAt: clerkOrganization.createdAt.toString(),
-  updatedAt: clerkOrganization.updatedAt.toString(),
-  apiServiceLevel: clerkOrganization.publicMetadata.apiServiceLevel,
-  defaultLocale: clerkOrganization.publicMetadata.defaultLocale,
-  domain: clerkOrganization.publicMetadata.domain,
-  productUseCases: clerkOrganization.publicMetadata.productUseCases,
-});
+const toUserEntity = (clerkUser: UserResource): IUserEntity => {
+  /*
+   * When mapping to IUserEntity, we have 2 cases:
+   *  - user exists and has signed in
+   *  - user is signing up
+   *
+   * In the case where the user is still signing up, we are using the clerk identifier for the id.
+   * This however quickly gets update to the externalId (which is actually the novu internal entity
+   * identifier) that gets used further in the app. A few consumers that want to user this identifier
+   * before it is set to the internal value monitor the flag 'shouldMonitor' and only use them when
+   * the internal Novu id is present.
+   */
+
+  return {
+    _id: clerkUser.externalId ?? clerkUser.id,
+    firstName: clerkUser.firstName,
+    lastName: clerkUser.lastName,
+    email: clerkUser.emailAddresses[0].emailAddress,
+    profilePicture: clerkUser.imageUrl,
+    createdAt: clerkUser.createdAt?.toString() ?? '',
+    showOnBoarding: clerkUser.publicMetadata.showOnBoarding,
+    showOnBoardingTour: clerkUser.publicMetadata.showOnBoardingTour,
+    servicesHashes: clerkUser.publicMetadata.servicesHashes,
+    jobTitle: clerkUser.publicMetadata.jobTitle,
+    hasPassword: clerkUser.passwordEnabled,
+  };
+};
+
+const toOrganizationEntity = (clerkOrganization: OrganizationResource): IOrganizationEntity => {
+  /*
+   * When mapping to IOrganizationEntity, we have 2 cases:
+   *  - user exists and has signed in
+   *  - user is signing up
+   *
+   * In the case where the user is still signing up, we are using the clerk identifier for the id.
+   * This however quickly gets update to the .publicMetadata.externalOrgId (which is actually the novu
+   * internal entity identifier) that gets used further in the app. A few consumers that want to user
+   * this identifier before it is set to the internal value monitor the flag 'shouldMonitor' and only
+   * use them when the internal Novu id is present.
+   */
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
👏 **praise:** Great commentary!‏